rotkehlchen/chain/ethereum/modules/base_bridge/decoder.py

```diff
@@ -0,0 +1,165 @@
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.eth = A_ETH.resolve_to_crypto_asset()
+
+    def _find_event(
+            self,
+            context: DecoderContext,
+            contract_address: ChecksumEvmAddress,
+            deposit_event: bytes,
+            asset: AssetWithSymbol,
+            from_address: ChecksumEvmAddress,
+            to_address: ChecksumEvmAddress,
+            amount: Optional[FVal] = None,
+    ) -> None:
+        """Given the filters for the events update the information of the bridging events"""
+        expected_event_type, new_event_type, from_chain, to_chain, expected_location_label = bridge_prepare_data(  # noqa: E501
+            tx_log=context.tx_log,
+            deposit_events=(deposit_event,),
+            main_chain=ChainID.ETHEREUM,
+            l2_chain=ChainID.BASE,
+            from_address=from_address,
+            to_address=to_address,
+        )
+
+        # Find the corresponding transfer event and update it
+        for event in context.decoded_events:
+            if (
+                event.event_type == expected_event_type and
+                event.location_label == expected_location_label and
+                event.address == contract_address and
+                event.asset == asset and
+                (amount is None or event.balance.amount == amount)
+            ):
+                bridge_match_transfer(
+                    event=event,
+                    from_address=from_address,
+                    to_address=to_address,
+                    from_chain=from_chain,
+                    to_chain=to_chain,
+                    amount=event.balance.amount,
+                    asset=asset,
+                    expected_event_type=expected_event_type,
+                    new_event_type=new_event_type,
+                    counterparty=BASE_CPT_DETAILS,
+                )
+
+    def _decode_bridge_eth(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a bridging event for ETH. Either a deposit or a withdrawal."""
+        if context.tx_log.topics[0] not in {TRANSACTION_DEPOSITED, WITHDRAWAL_FINALIZED}:
+            # Make sure that we are decoding a supported event.
+            return DEFAULT_DECODING_OUTPUT
+
+        # Read information from event's topics & data
+        if context.tx_log.topics[0] == TRANSACTION_DEPOSITED:
+            from_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+            to_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+        else:
+            from_address = context.transaction.from_address
+            to_address = context.transaction.from_address
+
```

Comment:
Maybe use a variable for context.transaction.from_address since it is used twice here? Not a big deal of course.

Reply:
mmm I believe it is not needed since you will do an extra memory pointer for no reason